### ISSUE TITLE
[data node] Spike graphql list in addition to cursor-based pagination

### ISSUE THREAD

**Issue by edd** [vegaprotocol/vega#6282]
# tl;dr
Some people find the [cursor-based pagination](https://graphql.org/learn/pagination/) implemented in the newer GraphQL calls to be overly complex. There might be an ‘escape hatch’ that lets us support simpler queries for those people, without significantly changing the API or losing the new pagination functionality.

# Longer intro
There has been a GraphQL API for Vega for ages. The way it was built originally favoured simplicity that scaled for a little while, but was somewhat limited in supporting actual historical data queries efficiently. So new queries were added (e.g. `assets` Is marked as `@deprecated` in favour of `assetsConnection`, which supports pagination. But for people building simple integrations, it adds the requirement to parse through sub properties - `edges { node { id } }` rather than just `{ id }`. As GraphQL is a tool used a lot for simple queries (‘Oh just go and fetch the assets from here’), losing that simple query structure has made some people sad.

## What if I told you we could have both?
🎤 ⬇️ 

OK, It’s not actually that simple. 

## Background
Star Wars API is the canonical GraphQL example doc. It’s like the pet store for Swagger/OpenAPI. It also has cursor based pagination, but for ease of use it also has this:
https://github.com/graphql/swapi-graphql/blob/6bb1fb84e3e210474980f42a0aff6e704d27187c/schema.graphql#L112-L137

What that allows is people to do simple queries:
```graphql

{
  allFilms {
    films {
      title
    }
  }
}

```

But also paginated ones:
```graphql
{
  allFilms {
    edges {
      node {
        title
      }
    }
  }
}
```

# Upsides
- It will help the people who use GraphQL for data exploration
- It’s a relatively minor schema change

# Downsides
- The nested `films` property accepts no parameters, so it’s pretty dumb. That will be fine for some queries (let’s say assets) but not for others (let’s say parties). Some data types will have huge amounts of data that will be unreasonable to send, or let people query for. So limits will need to be in place
  - What are those limits? Dunno
  - Is it bad developer experience to offer that sub property only on some connections? Dunno
 
# Spike
Let’s try it out for some simple use cases before putting to much time in to it. Let’s say market and assets?

**Comment by pscott31** [comment]
I think in summary what I was trying to say above is that - if you just want a shortcut to get the nodes out of a connection as an extra field (basically bypassing edges and the cursor for each edge); I am pretty happy with that and it should be quite easy to implement.

If we want to start adding extra queries or resolvers or whatever that do the pagination and really fetch all the data and stitch it together, I think I would require a fair bit more convincing

**Comment by edd** [comment]
> My other thought about this is it's getting a bit late in the day for big API changes. I don't love the extra typing but at least its consistent

I'm on your side here - the proposal to **try this out** was because I saw an approachthat *might* make the vocally anti-edges people happy. So unless someone else is willing to take up the cause, given your reluctance, I'm alright if this gets closed as `wontfix`.

**Comment by edd** [comment]
I talked this over with Dex and I'll close this proposal. His preference would be new queries that are simpler, which is a different (more straightforward in some ways) proposition. Let's get to that later.
